### PR TITLE
Add Foundation stake bot parameters

### PR DESCRIPTION
**Pull request by danpaul000**
Move Foundation program parameters out of CI infrastructure and into the open.

### CODE REVIEW

**Comment by danpaul000** [comment]
CI test (sans `--confirm`) looks good: https://buildkite.com/solana-labs/stake-o-matic/builds/621

File: stake-o-matic.sh [hunk 1]
```diff
@@ -15,7 +54,14 @@ if [[ $BUILDKITE = true ]]; then
   git config --global user.name "Solana Maintainers"
 fi
 
-./solana-stake-o-matic "$@"
+if [[ $CLUSTER == "testnet" ]]; then
+  ./solana-stake-o-matic "${TESTNET_ARGS[@]}"
+elif [[ $CLUSTER == "mainnet beta" ]]; then
```
Comment (mvines) [inline stake-o-matic.sh:59]: IMO kabab this, so mainnet-beta.  We never use "mainnet beta" in the code anywhere

File: stake-o-matic.sh [hunk 2]
```diff
@@ -6,6 +6,45 @@ set -ex
...
+  --max-infrastructure-concentration 25
+  --min-epoch-credit-percentage-of-average 35
+  --infrastructure-concentration-affects destake-new
+  --min-release-version 1.7.3
+  --markdown
+  $CONFIRM
+  stake-pool-v0
+  --baseline-stake-amount 5000
+  $RESERVE_ACCOUNT_ADDRESS
+  $STAKE_AUTHORITY_KEYPAIR
```
Comment (mvines) [inline stake-o-matic.sh:24]:
```suggestion
  ${RESERVE_ACCOUNT_ADDRESS:?}
  ${STAKE_AUTHORITY_KEYPAIR:?}
```

File: stake-o-matic.sh [hunk 3]
```diff
@@ -6,6 +6,45 @@ set -ex
 
 "$(dirname "$0")"/fetch-release.sh "$STAKE_O_MATIC_RELEASE"
 
+# shellcheck disable=SC2206
+TESTNET_ARGS=(
+  --url $URL
```
Comment (mvines) [inline stake-o-matic.sh:11]:
```suggestion
  --url ${URL:?}
```

File: stake-o-matic.sh [hunk 4]
```diff
@@ -6,6 +6,45 @@ set -ex
...
+  $CONFIRM
+  stake-pool-v0
+  --baseline-stake-amount 5000
+  $RESERVE_ACCOUNT_ADDRESS
+  $STAKE_AUTHORITY_KEYPAIR
+)
+
+# shellcheck disable=SC2206
+MAINNET_BETA_ARGS=(
+  --url $URL
```
Comment (mvines) [inline stake-o-matic.sh:29]:
```suggestion
  --url ${URL:?}
```

File: stake-o-matic.sh [hunk 5]
```diff
@@ -6,6 +6,45 @@ set -ex
...
+  --max-infrastructure-concentration 10
+  --infrastructure-concentration-affects destake-new
+  --min-self-stake 100
+  --min-testnet-participation 5 10
+  --markdown
+  $CONFIRM
+  stake-pool-v0
+  --baseline-stake-amount 25000
+  $RESERVE_ACCOUNT_ADDRESS
+  $STAKE_AUTHORITY_KEYPAIR
```
Comment (mvines) [inline stake-o-matic.sh:46]:
```suggestion
  ${RESERVE_ACCOUNT_ADDRESS:?}
  ${STAKE_AUTHORITY_KEYPAIR:?}
```

**Review by mvines — APPROVED** [review]
LGTM, with the nits to make bash fail the command if the required env vars aren't setup.

For the follower builds (perhaps as a different PR), we could add new env var that:
1. Adds `--require-classification`
2. Removes `--markdown`
3. Removes the `git push` at the end of the script.

**Pull request merged**